Documentation/MigrationGuide.md

```diff
@@ -0,0 +1,60 @@
+## SwiftGen 5.0 Migration Guide
+
+If you're migrating from Swift 4.x to SwiftGen 5.0, there might be some migration steps you'll need to use.
+
+Below is a list of pointers to help you migrate to the new SwiftGen 5.0
+
+## Command Line invocation
+
+### `images` command has been renamed
+
+The `images` subcommand has been renamed `xcassets`.
+You should replace invocations of `swiftgen images …` by `swiftgen xcassets …`
+
+### `--enumName` flag migrated to `--param`
+
+The `--enumName` flag is no longer supported.
+Instead, you can now pass arbitrary parameters to your templates using the `--param X=Y` flag.
+
+All the templates bundled in SwiftGen accept at least the `enumName` parameter to provide at least the same customization as before.
+
+So if you used `swiftgen <command> --enumName Foo` when invoking SwiftGen before, you should now invoke it via `swiftgen <command> --param enumName=Foo`
+
+### You're required to specify a template name or path (no more default template)
+
+Swift evolves so fast that deciding which template should be declared as the default one would require us to change the default template every time a new major version of Swift is released. Besides, our default might not be everyone's default. That's why we decided to not fallback to a default template anymore.
+
+Instead, you're now **required** to provide a template when invoking SwiftGen, either via `-t <templatename>` (you can see the list of available template names using the `swiftgen templates list` command) or via `-p <templatepath>`.
+
+ℹ️ The templates that were used as default template in SwiftGen 4.0 are still bundled with SwiftGen, but they have been renamed (mainly to `"swift2"` since the "default" template in SwiftGen 4 was the one we wrote back in Swift 2 times!) and you now have to explicitly tell which to use. See [below](#templates-bundled-with-swiftgen--some-renaming) for more info.
+
+✅ **Most of you probably already use the `-t swift3` option** if you were writing Swift3 code, so **you won't need to change anything**. But in the unlikely event that you were still writing Swift2 and didn't specify a template, you'll then have to use `-t swift2` to force using this (previously default) template.
+
+## Bundled Templates have been renamed & cleaned
+
+Some templates bundled with SwiftGen have been **renamed** to have a better naming consistency.
```

Comment:
Mention that others have been merged, or removed altogether. For example the storyboards ios/macos templates, or the "all-values"  and "no-comments" templates.